### ISSUE TITLE
Unknown object where spacesuit should be

### ISSUE THREAD

**Issue by sloantothebone** [Amaz1/moontest#3]
The moontest:spacesuit is gone

**Comment by fishyWET** [comment]
Feature, Old Spacesuit is removed, New is added in with a improved system. https://github.com/Amaz1/moontest/commit/0a99fc91a8b4b1efbb512d2ef5c0adc23d8d8f66

**Comment by sloantothebone** [comment]
Well then can you remove the original spacesuit from the inventory and replace it with a spacesuit from the improved mod and 50 oxxygen tanks?

**Comment by asl97** [comment]
it seem to have gotten re-added for some reason, https://github.com/Amaz1/moontest/commit/9c5363f691ec11a2010730dbeeb5caf32c01f0c8

**Comment by Amaz1** [comment]
Yeah, I re added it because the armour was not ideal... It ran out after to short a time, and you didn't get it right away, and so on and so forth.. If you want to remedy this, feel free to make a pull.

**Comment by neinwhal** [comment]
The armour were not suppose to be the air support, the air tanks are.
They would wear out rather slowly as i recalled testing _as long as_ the player doesn't not obtain any damage, which severely wears out the air tank, following the concept of how fragile air tanks are and how easily they can be punctured, forcing the player to make cautious steps.
The armours are only to be used in the last resort, acting as the last form of air supply, like how spacesuits in real life works, activating the last limited air supply.
As there is no good way to make the last amount of air supply "limited", the air supply will only renew when the player take damage, thus slowly draining the player health and hence causing an "limited air supply".

My contribution of the spacesuit armours was an attempt to make a realistic approach to moontest.
But i noted that many players will not appreciate the complexity of this (adding the fact that i did not provide much documentation on it), and hence the reason why i asked for the re-addition of the original spacesuit.

**Comment by Amaz1** [comment]
Ah, okay. I didn't realise that was the case :)

**Comment by neinwhal** [comment]
Alright i did a test.
This is with the assumption that the player is provided with the weakest spacesuit, skylab A7LB and the lowest capacity air tank, the steel air tank on starting the game.
The steel air tank (**lowest capacity/weakest air tank**) will last for **21 minutes 45 seconds** exactly.
I was able to find enough iron lumps and phosphorus lumps to cook them into steel ingots within the 10th minute with the initial starting items as provided by the game. And was able to explore the moon for the remaining 11 minutes.
Should i accidentally puncture my air tank/ forget to bring it along in the trip. My skylab A7LB supplied oxygen for 1 minute 30 seconds before wearing out, enough time for me to move across 314 nodes even with the spacesuit slowing me down.

This spacesuit imo, provides moontest with a survival feature, rather than just moving across the landscape with no restrictions. Of course, if there are players that prefer the later, they can simply switch to the old space suit.

I understand that the complexity of how the spacesuit works is not properly explained by me so i will provide a simple explanation here:

Spacesuits - Acts as armor and emergency air supply.
- From weakest to strongest - Skylab A7LB, Apollo A7LB, Baseline EMU, Enhanced EMU
- The strongest spacesuit provides more emergency air supply that will last the player longer but the slows down the player most.
- Note that air supply, or air bubbles in minetest context is only provided when the player starts drowning. The stronger spacesuit will give more bubbles when the player is detected drowning, while the weakest spacesuit will only give one air bubble.
- More info: https://github.com/Amaz1/moontest/blob/master/mods/spacesuit/init.lua#L6-L35

Oxygen tanks - Provide air for the player (Require a full set of spacesuit equipped to work)
- From weakest to strongest:  Steel Oxygen Tank, Titanium Oxygen Tank, Mese Oxygen Tank.
- Oxygen tanks fill the player's breath once it reaches 3 air bubbles, when the player's breath is renewed, the oxygen tanks also wears.

How to equip the spacesuits/ oxygen tanks;
![image](https://cloud.githubusercontent.com/assets/3191240/10121670/5e22953c-652a-11e5-9234-85ca290c482d.png)

**Comment by HybridDog** [comment]
Where does the oxygen come from?

**Comment by asl97** [comment]
we need pumps for filling the tanks :D

**Comment by HybridDog** [comment]
pumps and valves
